imagetest/test_suites/metadata/startup_shutdown_script_test.go

```diff
@@ -0,0 +1,127 @@
+//go:build cit
+// +build cit
+
+package metadata
+
+import (
+	"io/ioutil"
+	"os/exec"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+)
+
+// The designed shutdown limit is 90s. Let's verify it's executed no less than 80s.
+const shutdownTime = 80
+
+// TestStartupShutdownScripts tests that all startup/shutdown scripts
+// appropriate for the platform ran successfully.
+func TestStartupShutdownScripts(t *testing.T) {
+	isWindows := runtime.GOOS == "windows"
+	for _, ms := range metadataScripts {
+		if ms.windows != isWindows {
+			continue
+		}
+		bytes, err := ioutil.ReadFile(ms.outputPath)
+		if err != nil {
+			t.Fatalf("failed to read %s output %v", ms.outputPath, err)
+		}
+		output := strings.TrimSpace(string(bytes))
+		output = strings.Trim(output, `"`)
+		if output != ms.outputContent {
+			t.Fatalf(`%s output expected "%s", but actual "%s"`, ms.description, ms.outputContent, output)
+		}
+	}
+}
+
+// TestStartupShutdownScriptFailed tests that a script that fails to execute
+// doesn't crash the vm.
+func TestStartupShutdownScriptFailed(t *testing.T) {
+	for _, script := range metadataScripts {
+		if _, err := utils.GetMetadataAttribute(script.metadataKey); err != nil {
+			t.Fatalf("Couldn't get %s from metadata: %v", script.metadataKey, err)
+		}
+	}
+}
+
+// TestWindowsScriptOrder tests that Windows startup & shutdown scripts are
+// executed in the documented order.
+func TestWindowsScriptOrder(t *testing.T) {
+	if runtime.GOOS != "windows" {
+		t.Skip("Script run order only checked on Windows.")
+	}
+
+	var expectedStartupOrder, actualStartupOrder, expectedShutdownOrder, actualShutdownOrder []string
+
+	for _, ms := range metadataScripts {
+		if ms.windows {
+			outputFile := strings.TrimPrefix(ms.outputPath, "C:\\")
+			if strings.HasPrefix(outputFile, "shutdown") {
+				expectedShutdownOrder = append(expectedShutdownOrder, outputFile)
+			} else {
+				expectedStartupOrder = append(expectedStartupOrder, outputFile)
+			}
+		}
+	}
+
+	command := "Get-ChildItem \"C:\\*.txt\" | Sort-Object LastWriteTime | Select-Object Name"
+
+	output, err := utils.RunPowershellCmd(command)
+	if err != nil {
+		t.Fatalf("Cannot get output file list: %v - %s", err, output.Stderr)
+	}
+	outputSlice := strings.Split(output.Stdout, "\n")
+
+	for _, outputFile := range outputSlice {
+		outputFile = strings.TrimSpace(outputFile)
+		if strings.HasPrefix(outputFile, "shutdown") {
+			actualShutdownOrder = append(actualShutdownOrder, outputFile)
+		} else if strings.HasPrefix(outputFile, "startup") {
+			actualStartupOrder = append(actualStartupOrder, outputFile)
+		} else if strings.HasPrefix(outputFile, "sysprep") {
+			actualStartupOrder = append(actualStartupOrder, outputFile)
```

Comment:
most cases of repeated if/elseif in go are best written using switch/case

Reply:
Done.